### ISSUE TITLE
AML modifies sound by itself

### ISSUE THREAD

**Issue by MPeti1** [Zackptg5/Audio-Modification-Library#2]
Once I had this problem, but one of the dev's sad that the bug is not in AML.
Now the problem is here again, started yesterday, all sound mods disabled in Magisk Manager, and the sound is still modified when AML is enabled. I tried to uninstall AML, reboot, and install it again, but it's the same. I can't imagine it's not caused by AML somehow

I have 4 sound mods installed:
- Viper4AndroidFX ([link](https://forum.xda-developers.com/apps/magisk/module-viper4android-fx-2-5-0-5-t3577058))
  - version 2.5.0.5
  - old themed app
  - 64 bit driver
- Dolby Atmos ([link](https://forum.xda-developers.com/android/apps-games/soundmod-axon-7-dolby-atmos-t3412342))
  - mode: Lenovo A7000
- JamesDSP ([link](https://forum.xda-developers.com/android/software/soundmod-ainur-audio-t3450516))
  - new patch method
  - not huawei
- Audio Compatibility Patch ([link](https://forum.xda-developers.com/apps/magisk/module-universal-deepbuffer-remover-t3577067))

All disabled since I started testing which is malfunctioning, and because they amplify that bad sound.

Since AML's installer logs the detected and patched sound modifiers, I know that I disabled all sound modification modules.

A strange thing is that the bug is exactly the same: in the morning when I got up I noticed that the system sounds also got modified, and they got very noisy and rough. Please note that I've installed Audio Compatibility Patch a few days and reboots before that strange behavior, and in that time everything worked perfectly.

The list of all installed modules:
![](https://i.imgur.com/gHgLYuJ.png)

Please note that LibFS is a custom module, and it just puts 2 files to /system/bin

My phone is a Xiaomi Redmi 3S Prime with MIUI 8.
Magisk version is 17.1, Manager version is 6.0.1 (151)
All modules are up to date, exept the topmost on the picture, it got a new update today.

Soon I'll also include a Magisk log.

**Comment by MPeti1** [comment]
Sorry for the late reply, here are the logs:
[logcat.txt](https://github.com/Zackptg5/Audio-Modification-Library/files/2549452/logcat.txt)
[magisk_log_20181105_075731.log](https://github.com/Zackptg5/Audio-Modification-Library/files/2549453/magisk_log_20181105_075731.log)

Please note that before I exported the logs I uninstalled ACP module.

**Comment by Zackptg5** [comment]
So if you want to see what AML module is, just go to /sbin/.core/img/aml in any root file explorer where you can see hidden folders. All that aml contains are config, policy, and mixer files along with a boot script and a system.prop file that contain all of the patches from detected audio mods. So:

In the case of app/lib containing mods like v4a, dolby, and jdsp, when you disable those mods but leave AML enabled (not sure why you would do this), the actual stuff modifying sound (the app and libs) are not mounted since they're still contained in the v4a, dolby and jdsp modules. You may get some errors in the logcat about it though since the configs still mounted by aml are expecting v4a, dolby, and jdsp libs which don't exist since those mods aren't mounted.

ACP is the weird one in this case. All ACP is is some policy patches. These get pulled into aml and so the actual acp module in this case would contain pretty much nothing. AML has all of ACP patches at this point. So if you disable acp but leave aml enabled, the acp patches are still active and doing their thing. However ACP it itself doesn't modify the "sound", it just modifies how android routes audio.

So long story short, there's nothing in aml that will modify how things sound, it's just patches and props.

The big question here is why do you have aml enabled when the audio mods it's patching are disabled? That's just asking for trouble and defeats the purpose of having it. If you disable the audio mods, you should disable aml too. 

I'll try to add something to aml in the future to have it remove the patches for the audio mods that are disabled (I have it set to only remove the patches for audio mods that are uninstalled currently)

**Comment by Zackptg5** [comment]
Added the disable stuff to aml, see commit e430a17

**Comment by MPeti1** [comment]
I've disabled the mods (except AML) to see if I still hear that strange effect that came a few days ago. I've done that with all of the installed sound mods: disabled all (except AML), and came to the conclusion that nothing changes with the disabling of any of them. But it's good to know that AML loads the configs of disabled mods, and afterwards it is logical, but for some reason I thought that if I disable a sound mod it'll just be disabled.
I was unsure about the source of that effect because I did not modify any mods' config, not installed any new sound mod, not updated, enabled or disabled any of them, and since I've installed ACP just a few days before that strange effect appeared I thought that mod can be the cause, and after I've uninstalled it I thought AML could have persisted some remnants of ACP, and that's why that effect is still there.

Now I've reinstalled all sound mods, and everything seems to be fine :D Although I don't know why was it enough, if it weren't before
